website/src/controllers/api-v3/groups.js

```diff
@@ -391,145 +395,105 @@ api.removeGroupMember = {
-          }
-
-          cb();
-        });
-      }
-    });
-  }, function(err){
-    if(err) return err.code ? res.json(err.code, {err: err.err}) : next(err);
-
-    async.series([
-      function(cb) {
-        group.save(cb);
-      },
-      function(cb) {
-        // TODO pass group from save above don't find it again, or you have to find it again in order to run populate?
-        populateQuery(group.type, Group.findById(group._id)).exec(function(err, populatedGroup){
-          if(err) return next(err);
-
-          res.json(populatedGroup);
-        });
-      }
-    ]);
-  });
-};
+  if (!userToInvite) {
+    throw new NotFound(res.t('userWithIDNotFound', {userId: uuid}));
+  }
 
-function _inviteByEmails (emails, group, inviter, req, res, next) {
-  let usersAlreadyRegistered = [];
-  let invitesToSend = [];
-
-  return Q.all(emails.forEach(invite => {
-    if (!invite.email) throw new BadRequest(res.t('inviteMissingEmail'));
-
-    return User.findOne({$or: [
-      {'auth.local.email': invite.email},
-      {'auth.facebook.emails.value': invite.email}
-    ]})
-    .select({_id: true, 'preferences.emailNotifications': true})
-    .exec()
-    .then(userToContact => {
-      if(userToContact){
-        usersAlreadyRegistered.push(userToContact._id); // TODO does it work not returning
-      } else {
-        // yeah, it supports guild too but for backward compatibility we'll use partyInvite as query
-        // TODO absolutely refactor this horrible code
-        let link = `?partyInvite=${utils.encrypt(JSON.stringify({id: group._id, inviter: inviter, name: group.name}))}`;
-
-        let inviterVars = getUserInfo(inviter, ['name', 'email']);
-        let variables = [
-          {name: 'LINK', content: link},
-          {name: 'INVITER', content: req.body.inviter || inviterVars.name},
-          {name: 'REPLY_TO_ADDRESS', content: inviterVars.email}
-        ];
-
-        if(group.type == 'guild'){
-          variables.push({name: 'GUILD_NAME', content: group.name});
-        }
-
-        // TODO implement "users can only be invited once"
-        // Check for the email address not to be unsubscribed
-        return EmailUnsubscription.findOne({email: invite.email}).exec()
-        .then(unsubscribed => {
-          if (!unsubscribed) utils.txnEmail(invite, ('invite-friend' + (group.type == 'guild' ? '-guild' : '')), variables);
-        });
-      }
-    });
-  }))
-  .then(() => {
-    if (usersAlreadyRegistered.length > 0){
-      return _inviteByUUIDs(usersAlreadyRegistered, group, inviter, req, res, next);
+  if (group.type === 'guild') {
+    if (_.contains(userToInvite.guilds, group._id)) {
+      throw new NotAuthorized(res.t('userAlreadyInGroup'));
+    }
+    if (_.find(userToInvite.invitations.guilds, {id: group._id})) {
+      throw new NotAuthorized(res.t('userAlreadyInvitedToGroup'));
+    }
+    userToInvite.invitations.guilds.push({id: group._id, name: group.name, inviter: res.locals.user._id});
```

Comment:
@TheHollidayInn instead of `inviter: res.locals.user._id` you could use `inviter: inviter._id`